edge/cmd/edgecore/app/server.go

```diff
@@ -62,6 +65,31 @@ offering HTTP client capabilities to components of cloud to reach HTTP servers r
 			if err != nil {
 				klog.Exit(err)
 			}
+
+			// should not save token in config file
+			if config.Modules.EdgeHub.Token != "" {
+				go func() {
+					// if receive data from CleanupTokenChan
+					// it means that edgecore apply for ca/certs successfully, then we can cleanup token
+					<-certificate.CleanupTokenChan
+
+					// cleanup token
+					if err := cleanupToken(*config, opts.ConfigFile); err != nil {
+						klog.Exit(err)
+					}
+				}()
+			}
+
+			// Get token from bootstrap file if it exists
+			bootstrapFile := "/etc/kubeedge/bootstrap-edgecore.conf"
+			if utilvalidation.FileIsExist(bootstrapFile) {
```

Comment:
if the file does not exist, we should throw an err and exitt

Reply:
if edgecore restart, bootstrap file also not exist, we should not throw an error

Reply:
add a pre check